src/test/groovy/com/palantir/gradle/revapi/RevapiSpec.groovy

```diff
@@ -852,6 +853,133 @@ class RevapiSpec extends IntegrationSpec {
         println runTasksSuccessfully('revapi').standardOutput
     }
 
+    def 'changing a protected method in an immutables class is not a break'() {
+        when:
+        rootProjectNameIs('root')
+
+        buildFile << """
+            apply plugin: '${TestConstants.PLUGIN_NAME}'
+            apply plugin: 'java'
+            apply plugin: 'maven-publish'
+            
+            allprojects {
+                group = 'revapi.test'
+                ${mavenRepoGradle()}
+            }
+            
+            version = '1.0.0'
+            
+            revapi {
+                oldVersion = project.version
+            }
+
+            repositories {
+                mavenCentral()
+            }
+
+            dependencies {
+                annotationProcessor "org.immutables:value:2.8.8"
+                compileOnly "org.immutables:value:2.8.8:annotations"  
+            }
+            
+            ${testMavenPublication()}
+        """.stripIndent()
+
+        def methodChanges = [
+                new MethodChange(
+                        'protected abstract String returnTypeChangedProtectedParam();',
+                        'protected abstract int returnTypeChangedProtectedParam();',
+                        false),
+                new MethodChange(
+                        'public abstract String returnTypeChangedPublicParam();',
+                        'public abstract int returnTypeChangedPublicParam();',
+                        true),
+                new MethodChange(
+                        '',
+                        'public abstract String newPublicParam();',
+                        false),
```

Comment:
Think this should break, unless it's Optional / a container / non-abstract, since clients not setting this would start getting runtime errors when constructing this object.

Reply:
That's not a compile/ABI break though - you could argue it's expanding the scope of revapi quite a bit.

Reply:
I guess it would need to be a brand new check in order to give a good error message.